### PR TITLE
Sprinttwo

### PR DESCRIPTION
**Pull request by girishchandrad**
Hi Juan,

Please review ember sprint two.

thanks,

GC

### CODE REVIEW

File: app/controllers/application.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Controller from '@ember/controller';
+
+export default Controller.extend({
+  routing: Ember.inject.service('-routing'),
+  currentRouteName: Ember.computed.reads('routing.currentRouteName'),
+  checkRoute: Ember.computed.match('currentRouteName', /index/),
```
Comment (juandsierra82) [inline app/controllers/application.js]: contorllers do not need to have a routing computed property. please check on emberjs docs on how to access the route object from a controller. see if these methods can be accessed directly by the controller using controller methods.

File: app/routes/form.js [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+    "name":"541",
+    "link":"https://www.ftb.ca.gov/forms/2017/17_541.pdf#",
+    "instructionsLink":"https://www.ftb.ca.gov/forms/2017/17_541bk.pdf",
+    "efileRrequiredDesc":"Yes",
+    "efileAllowedDesc":"Depends",
+    "dd":"04/15/2018",
+    "fiscalDdDesc":"Five months after fiscal year end",
+    "extDd":"09/15/2018",
+    "fiscalExtDdDesc":"Five months after fiscal year end"}
+  }
```
Comment (juandsierra82) [inline app/routes/form.js:18]: One thing, does the spreadsheet have a short description of the form somewhere? if not that's cool.

File: app/routes/search.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  {"id":"333",
+  "jurisdiction":"California (CA)","entity-type":"S-Corp","desc":"California S Corporation Franchise or Income Tax Return ","name":"100-S","link":"https://www.ftb.ca.gov/forms/2017/17_100s.pdf#","instructions-link":"https://www.ftb.ca.gov/forms/2017/17_100sbk.pdf","efile-allowed":null,"efile-required":null,"efile-required-desc":"Yes","efile-allowed-desc":"Depends","dd":"04/15/2018","fiscal-dd-desc":"File Form 100S by the 15th day of the 3rd month after the close of the taxable year unless the return is for a short-period as required under R\u0026TC Section 24634","ext-dd":"09/15/2018","fiscal-ext-dd-desc":"it may file on or before the 15th day of the 9th month without filing a written request for an extension","created-at":"2018-05-02T14:22:06.020Z","updated-at":"2018-05-02T14:22:06.020Z"}],
+
+  model(){
+    return   this.get('data')
```
Comment (juandsierra82) [inline app/routes/search.js]: dont forget ;

File: app/styles/site-layouts.scss [hunk 4]
```diff
@@ -1,3 +1,22 @@
-body {
- padding: 10px;
+$app-qgreen: rgba(98,144,128,0.25);
+$app-halfgreen: rgba(98,144,128,0.5);
```
Comment (juandsierra82) [inline app/styles/site-layouts.scss]: constants need to be saved in the constants style sheet file i believe it's called constants.scss or you can do it in  a helpers.scss files. once they're there you can import them in nested sheets.

File: app/styles/site-layouts.scss [hunk 5]
```diff
@@ -1,3 +1,22 @@
...
+
+.search-button {
+  color: $app-halfgreen;
 }
+
+.text-icon {
+  text-decoration-color: $app-qgreen;
+  font-size: 31.5px;
```
Comment (juandsierra82) [inline app/styles/site-layouts.scss]: try to save font-sizes as constants too.

File: app/styles/site-views/form-view.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+.centerdiv-form-wrapper {
+  margin-top: 20px;
+  margin-bottom: 40px;
+}
+
+.centereddiv-form {
+  height: auto;
+  width:auto;
+  // background-color: #000;
+}
```
Comment (juandsierra82) [inline app/styles/site-views/form-view.scss]: delete the property if it won't be used. don't just comment it out.

File: app/styles/site-views/search-list.scss [hunk 7]
```diff
@@ -0,0 +1,80 @@
+$app-qgreen: rgba(98,144,128,0.25);
+$app-halfgreen: rgba(98,144,128,0.5);
+$app-lightgray: rgb(239,239,239);
+
+.centerdiv-list-wrapper {
+  background-color: $app-lightgray;
+}
+
+.centereddiv-list {
+
```
Comment (juandsierra82) [inline app/styles/site-views/search-list.scss]: centereddiv-list should be a different name. classes always have names in this format: first-second-third (with - ) between. also no need to say center-div just call it center-list

File: app/styles/site-views/search-list.scss [hunk 8]
```diff
@@ -0,0 +1,80 @@
+$app-qgreen: rgba(98,144,128,0.25);
+$app-halfgreen: rgba(98,144,128,0.5);
+$app-lightgray: rgb(239,239,239);
+
+.centerdiv-list-wrapper {
+  background-color: $app-lightgray;
```
Comment (juandsierra82) [inline app/styles/site-views/search-list.scss]: if this is a wrapper that means that all of its children should be nested inside this class in the .scss style sheet: .parent { .child { .grand-child { }}}
the nesting helps keep your classes from being duplicated or changed by subsequent style sheets. please nest when appropriate.

File: app/styles/site-views/search-list.scss [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+.list-footer {
+  font-size: 15px;
+  margin-top: 20px;
+}
+
+.list-toprow {
+  margin-top: 5px;
+
```
Comment (juandsierra82) [inline app/styles/site-views/search-list.scss]: same as a above -.list-top-row (these are custom classes right? not overriding classes in bootstrap?)

File: app/styles/site-views/search-list.scss [hunk 10]
```diff
@@ -0,0 +1,80 @@
+$app-qgreen: rgba(98,144,128,0.25);
```
Comment (juandsierra82) [inline app/styles/site-views/search-list.scss]: please look below at my comments and review the nesting of child components in parent components. http://sass-lang.com/documentation/file.SASS_REFERENCE.html#nested_rules

File: app/templates/application.hbs [hunk 11]
```diff
@@ -1,19 +1,21 @@
-<div style="height: 350px;">
-  <div class="d-flex align-items-center flex-column justify-content-center h-100 shadow p-3 mb-5 bg-white rounded">
-    <div><img class="pull-left img-thumbnail " src="images/firm_initials.png">
-      <span class="display-5 font-weight-bold text-muted " >TAX DUE DATE SEARCH</span>
+{{#if checkRoute}}
```
Comment (juandsierra82) [inline app/templates/application.hbs]: be sure to review my comments on the controller on how to do this using native controller methods. you will probably still need a computed property like this here though.

File: app/templates/form.hbs [hunk 12]
```diff
@@ -1,78 +1,79 @@
...
-    Connecticut (CT) - Form 20C
+    {{model.jurisdiction}}-{{model.name}}
   </h4>
   <br />
   <form >
     <div class="form-group row">
       <label for="" class="col-sm-2 form-control-label font-weight-bold ">Jurisdiction:</label>
       <div class="col-sm-4">
-        <span class="font-italic">Connecticut (CT)</span>
+        <span class="ffont-italic text-info text-info">{{model.jurisdiction}}</span>
```
Comment (juandsierra82) [inline app/templates/form.hbs]: ffont-italic is a mystype? not sure double check please.... mmm reading some more i guess it's right?

File: app/templates/search.hbs [hunk 13]
```diff
@@ -0,0 +1,95 @@
+
```
Comment (juandsierra82) [inline app/templates/search.hbs:1]: you solved the little footer line problem. good job! how did you do it?

**Review by juandsierra82 — COMMENTED** [review]
so on searches the tax search input should be the same size as in the first view in terms of length.
- the left tab should be pushed so it's aligned with the filters, also decrease some of the padding on that top section
- on the form yorence had a good idea, the due date should be to the top right and the same font size as the form name

File: app/routes/search.js (hunk 3 above)
Comment (girishchandrad) [reply app/routes/search.js]: done

File: app/routes/form.js (hunk 2 above)
Comment (girishchandrad) [reply app/routes/form.js:18]: I will talk to Albert about these descriptions on Monday. For now it is what it is.

File: app/controllers/application.js (hunk 1 above)
Comment (girishchandrad) [reply app/controllers/application.js]: done. thanks for you help. Added ember truth-helper

File: app/styles/site-layouts.scss (hunk 4 above)
Comment (girishchandrad) [reply app/styles/site-layouts.scss]: done

File: app/styles/site-layouts.scss (hunk 5 above)
Comment (girishchandrad) [reply app/styles/site-layouts.scss]: done

File: app/styles/site-views/form-view.scss (hunk 6 above)
Comment (girishchandrad) [reply app/styles/site-views/form-view.scss]: removed

File: app/styles/site-views/search-list.scss (hunk 9 above)
Comment (girishchandrad) [reply app/styles/site-views/search-list.scss]: yes custom classes.

Comment (girishchandrad) [reply app/styles/site-views/search-list.scss]: i have changed classes with prefix such as landing, list or form.

File: app/styles/site-views/search-list.scss (hunk 10 above)
Comment (girishchandrad) [reply app/styles/site-views/search-list.scss]: done

File: app/templates/form.hbs (hunk 12 above)
Comment (girishchandrad) [reply app/templates/form.hbs]: corrected

File: app/templates/search.hbs (hunk 13 above)
Comment (girishchandrad) [reply app/templates/search.hbs:1]: yes. I added a custom class and changed the font styles.

File: app/styles/site-views/search-list.scss (hunk 7 above)
Comment (girishchandrad) [reply app/styles/site-views/search-list.scss]: changed names

File: app/styles/site-views/search-list.scss (hunk 8 above)
Comment (girishchandrad) [reply app/styles/site-views/search-list.scss]: ok done.

**Pull request merged**